src/main/java/com/minecolonies/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -0,0 +1,325 @@
+    {
+        updateArmor();
+    }
+
+    /**
+     * Updates the equipment. Always take the first item of each type and set it.
+     */
+    protected void updateArmor()
+    {
+        worker.setItemStackToSlot(EntityEquipmentSlot.CHEST, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.FEET, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.HEAD, null);
+        worker.setItemStackToSlot(EntityEquipmentSlot.LEGS, null);
+
+        for (int i = 0; i < worker.getInventoryCitizen().getSizeInventory(); i++)
+        {
+            ItemStack stack = worker.getInventoryCitizen().getStackInSlot(i);
+
+            if (stack == null || stack.stackSize == 0)
+            {
+                worker.getInventoryCitizen().setInventorySlotContents(i, null);
+                continue;
+            }
+
+            if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == null)
+            {
+                worker.setItemStackToSlot(((ItemArmor) stack.getItem()).armorType, stack);
+            }
+        }
+    }
+
+    /**
+     * Goes back to the building and tries to take armour from it when he hasn't in his inventory.
+     *
+     * @return the next state to go to.
+     */
+    protected AIState goToBuilding()
+    {
+        if (!walkToBuilding())
+        {
+            final AbstractBuildingWorker workBuilding = getOwnBuilding();
+            if (workBuilding != null)
+            {
+                final TileEntityColonyBuilding chest = workBuilding.getTileEntity();
+
+                for (int i = 0; i < workBuilding.getTileEntity().getSizeInventory(); i++)
+                {
+                    final ItemStack stack = chest.getStackInSlot(i);
+
+                    if (stack == null)
+                    {
+                        continue;
+                    }
+
+                    if (stack.getItem() instanceof ItemArmor && worker.getItemStackFromSlot(((ItemArmor) stack.getItem()).armorType) == null)
+                    {
+                        final int emptySlot = worker.getInventoryCitizen().getFirstEmptySlot();
+
+                        if (emptySlot != -1)
+                        {
+                            worker.getInventoryCitizen().setInventorySlotContents(emptySlot, stack);
+                            chest.setInventorySlotContents(i, null);
+                        }
+                    }
+                    else if (!(stack.getItem() instanceof ItemBow || Utils.doesItemServeAsWeapon(stack)))
+                    {
+                        //todo dump everything which isn't a weapon or armor
+                    }
+                }
+            }
+            attacksExecuted = 0;
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        return AIState.GUARD_RESTOCK;
+    }
+
+    /**
+     * Chooses a target from the list.
+     *
+     * @return the next state.
+     */
+    protected AIState getTarget()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getTarget" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)